File: Data Structures & Algorithms/pow-x-n/submission-6.cpp
class Solution {
public:
    double myPow(double x, int n) {
        long long exp = n;
        if(exp < 0) {
            x = 1/x;
            exp = -exp;
        }
        double res = 1;
        double base = x;
        while(exp > 0){
            if(exp % 2 != 0){
                res *= base;
            }
            exp /= 2;
            base *= base;
        }
        return res;
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-6.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        count = {}
        l = 0
        res = 0
        for r in range(len(s)):
            if s[r] in count:
                l = max(count[s[r]] + 1, l)
            count[s[r]] = r
            res = max(res, r - l + 1)
        return res
